controlplane/src/core/bufservices/PlatformService.ts

```diff
@@ -3399,5 +3808,205 @@ export default function (opts: RouterOptions): Partial<ServiceImpl<typeof Platfo
+              break;
+            }
+            case 'Viewer': {
+              keycloakGroupName = `/${authContext.organizationSlug}/viewer`;
+              break;
+            }
+            default: {
+              throw new Error(`The role ${mapper.role} doesn't exist `);
+            }
+          }
+
+          await opts.keycloakClient.createIDPMapper({
+            realm: opts.keycloakRealm,
+            orgSlug: authContext.organizationSlug,
+            claims,
+            keycloakGroupName,
+          });
+        }
+
+        return {
+          response: {
+            code: EnumStatusCode.OK,
+          },
+          signInURL: `${opts.keycloakApiUrl}/realms/${opts.keycloakRealm}/broker/${authContext.organizationSlug}/endpoint`,
+          signOutURL: `${opts.keycloakApiUrl}/realms/${opts.keycloakRealm}/broker/${authContext.organizationSlug}/endpoint/logout_response`,
+        };
+      });
+    },
+
+    getOIDCProvider: (req, ctx) => {
+      const logger = opts.logger.child({
+        service: ctx.service.typeName,
+        method: ctx.method.name,
+      });
+
+      return handleError<PlainMessage<GetOIDCProviderResponse>>(logger, async () => {
+        const authContext = await opts.authenticator.authenticate(ctx.requestHeader);
+        const orgRepo = new OrganizationRepository(opts.db);
+
+        await opts.keycloakClient.authenticateClient();
+
+        const provider = await orgRepo.getOidcProvider({ organizationId: authContext.organizationId });
+        if (!provider) {
+          return {
+            response: {
+              code: EnumStatusCode.ERR_NOT_FOUND,
+            },
+            name: '',
+            endpoint: '',
+          };
+        }
+
+        return {
+          response: {
+            code: EnumStatusCode.OK,
+          },
+          name: provider.name,
+          endpoint: provider.endpoint,
+        };
+      });
+    },
+
+    deleteOIDCProvider: (req, ctx) => {
+      const logger = opts.logger.child({
+        service: ctx.service.typeName,
+        method: ctx.method.name,
+      });
+
+      return handleError<PlainMessage<DeleteOIDCProviderResponse>>(logger, async () => {
+        const authContext = await opts.authenticator.authenticate(ctx.requestHeader);
+        const orgRepo = new OrganizationRepository(opts.db);
+        const userRepo = new UserRepository(opts.db);
+
+        const hasAccess = await userRepo.checkUserAccess({
+          userID: authContext.userId,
+          organizationID: authContext.organizationId,
+          rolesToBe: ['admin', 'member'],
+        });
+
+        if (!hasAccess) {
+          return {
+            response: {
+              code: EnumStatusCode.ERR,
+              details: `The user doesnt have the permissions to perform this operation`,
+            },
+          };
+        }
+
+        await opts.keycloakClient.authenticateClient();
+
+        const organization = await orgRepo.byId(authContext.organizationId);
```

Comment:
Please put this logic into a service. We should try to prevent adding much logic here.

Reply:
Done